### ISSUE TITLE
Flaky row_sparse_pull test

### ISSUE THREAD

**Issue by eric-haibin-lin** [apache/incubator-mxnet#9384]
http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-9329-merge/21/pipeline

```
test_kvstore_gpu.test_row_sparse_pull ... /workspace/tests/ci_build/with_the_same_user: line 30: 122115 Segmentation fault      (core dumped) sudo -u "#${CI_BUILD_UID}" --preserve-env "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}" "HOME=${CI_BUILD_HOME}" ${COMMAND[@]}

script returned exit code 139
```

@reminisce

**Comment by marcoabreu** [comment]
Maybe related to #9378 ?

**Comment by eric-haibin-lin** [comment]
fixed via #11519
